src/tcl.c

```diff
@@ -965,3 +966,15 @@ int fork_before_tcl()
 #endif
   return 0;
 }
+
+time_t get_expire_time(Tcl_Interp * irp, const char *s) {
+  long expire_foo = strtol(s, NULL, 10);
+
+  if (expire_foo == 0)
```

Comment:
this needs more error checking, supply a ptr to strtol and verify it's \0 after parsing. we like to have an error if users supply not-a-number

Reply:
done. also added reject of negative numbers.